examples/scripts/deconv_ppp_bm4d_admm.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This file is part of the SCICO package. Details of the copyright
+# and user license can be found in the 'LICENSE.txt' file distributed
+# with the package.
+
+"""
+Image Deconvolution (ADMM Plug-and-Play Priors w/ BM4D)
+=======================================================
+
+This example demonstrates the use of class
+[admm.ADMM](../_autosummary/scico.optimize.rst#scico.optimize.ADMM) to
+solve a 3D image deconvolution problem using the Plug-and-Play Priors
+framework :cite:`venkatakrishnan-2013-plugandplay2`, using BM4D
+:cite:`maggioni-2012-nonlocal` as a denoiser.
+
+The 3D deconvolution problem is the task of recovering a 3D image
+that has been convolved with a 3D kernel and corrupted by noise.
+"""
+
+import numpy as np
+
+import jax
+
+import scico.numpy as snp
+from scico import functional, linop, loss, metric, plot, random
+from scico.examples import create_3D_foam_phantom, downsample_volume, tile_volume_slices
+from scico.optimize.admm import ADMM, LinearSubproblemSolver
+from scico.util import device_info
+
+"""
+Create a ground truth image.
+"""
+np.random.seed(1234)
+Nx = 128
+Ny = 128
+Nz = 128
+upsamp = 2
+x_gt_hires = create_3D_foam_phantom((upsamp * Nz, upsamp * Ny, upsamp * Nx), N_sphere=100)
+x_gt = downsample_volume(x_gt_hires, upsamp)
+x_gt = jax.device_put(x_gt)  # convert to jax array, push to GPU
+
+
+"""
+Set up forward operator and test signal consisting of blurred signal with
+additive Gaussian noise.
+"""
+n = 5  # convolution kernel size
+σ = 20.0 / 255  # noise level
+
+psf = snp.ones((n, n, n)) / (n ** 3)
+A = linop.Convolve(h=psf, input_shape=x_gt.shape)
+
+Ax = A(x_gt)  # blurred image
+noise, key = random.randn(Ax.shape)
+y = Ax + σ * noise
+
+
+"""
+Set up ADMM solver.
+"""
+f = loss.SquaredL2Loss(y=y, A=A)
+C = linop.Identity(x_gt.shape)
+
+λ = 20.0 / 255  # BM4D regularization strength
+g = λ * functional.BM4D()
+
+ρ = 1.0  # ADMM penalty parameter
```

Comment:
Any tuning here? No need to do a thorough search, but it may be worth doing a few manual runs to see if different values give better results?

Reply:
I did a manual tuning to improve the result slightly.